File: Python/programmers/etc/level3/거스름돈.py
def solution(n, money):
    dp = [0]*(n+1)
    dp[0] = 1
    for m in money:
        for i in range(m, n+1):
            dp[i] += dp[i-m]
    return dp[n]


n = 5
money = [1,2,5]
print(solution(n, money)) # 4

File: Java/programmers/etc/level3/거스름돈.java
package CodeTest.Java.programmers.etc.level3;

public class 거스름돈 {
    public int solution(int n, int[] money) {
        int[] dp = new int[n+1];
        dp[0] = 1;
        for (int m : money) {
            for (int i=m; i<n+1; i++) {
                dp[i] += dp[i-m];
            }
        }
        return dp[n];
    }
}
